crates/scroll/chainspec/src/lib.rs

```diff
@@ -0,0 +1,592 @@
+//! Scroll-Reth chain specs.
+
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/paradigmxyz/reth/main/assets/reth-docs.png",
+    html_favicon_url = "https://avatars0.githubusercontent.com/u/97369466?s=256",
+    issue_tracker_base_url = "https://github.com/paradigmxyz/reth/issues/"
+)]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+mod dev;
+mod genesis;
+mod scroll;
+mod scroll_sepolia;
+
+use alloc::{boxed::Box, vec::Vec};
+use alloy_chains::Chain;
+use alloy_consensus::Header;
+use alloy_genesis::Genesis;
+use alloy_primitives::{B256, U256};
+use derive_more::{Constructor, Deref, Display, From, Into};
+pub use dev::SCROLL_DEV;
+pub use genesis::ScrollChainInfo;
+#[cfg(not(feature = "std"))]
+pub(crate) use once_cell::sync::Lazy as LazyLock;
+use reth_chainspec::{
+    BaseFeeParams, ChainSpec, ChainSpecBuilder, DepositContract, EthChainSpec, EthereumHardforks,
+    ForkFilter, ForkId, Hardforks, Head,
+};
+use reth_ethereum_forks::{ChainHardforks, EthereumHardfork, ForkCondition, Hardfork};
+use reth_network_peers::NodeRecord;
+use reth_scroll_forks::ScrollHardforks;
+pub use scroll::SCROLL_MAINNET;
+pub use scroll_sepolia::SCROLL_SEPOLIA;
+#[cfg(feature = "std")]
+pub(crate) use std::sync::LazyLock;
+
+/// Chain spec builder for a Scroll chain.
+#[derive(Debug, Default, From)]
+pub struct ScrollChainSpecBuilder {
+    /// [`ChainSpecBuilder`]
+    inner: ChainSpecBuilder,
+}
+
+impl ScrollChainSpecBuilder {
+    /// Construct a new builder from the scroll mainnet chain spec.
+    pub fn scroll_mainnet() -> Self {
+        let mut inner = ChainSpecBuilder::default()
+            .chain(SCROLL_MAINNET.chain)
+            .genesis(SCROLL_MAINNET.genesis.clone());
+        let forks = SCROLL_MAINNET.hardforks.clone();
+        inner = inner.with_forks(forks);
+
+        Self { inner }
+    }
+}
+
+impl ScrollChainSpecBuilder {
+    /// Set the chain ID
+    pub fn chain(mut self, chain: Chain) -> Self {
+        self.inner = self.inner.chain(chain);
+        self
+    }
+
+    /// Set the genesis block.
+    pub fn genesis(mut self, genesis: Genesis) -> Self {
+        self.inner = self.inner.genesis(genesis);
+        self
+    }
+
+    /// Add the given fork with the given activation condition to the spec.
+    pub fn with_fork<H: Hardfork>(mut self, fork: H, condition: ForkCondition) -> Self {
+        self.inner = self.inner.with_fork(fork, condition);
+        self
+    }
+
+    /// Add the given forks with the given activation condition to the spec.
+    pub fn with_forks(mut self, forks: ChainHardforks) -> Self {
+        self.inner = self.inner.with_forks(forks);
+        self
+    }
+
+    /// Remove the given fork from the spec.
+    pub fn without_fork(mut self, fork: reth_scroll_forks::ScrollHardfork) -> Self {
+        self.inner = self.inner.without_fork(fork);
+        self
+    }
+
```

Comment:
I think you are missing a `archimedes_activated` which should call `self.inner.london_activated();`:
```rust
    self.inner = self.inner.london_activated();
    self.inner = self
            .inner
            .with_fork(reth_scroll_forks::ScrollHardfork::Archimedes, ForkCondition::Block(0));
```

Reply:
fixed [1a69d3a](https://github.com/scroll-tech/reth/pull/56/commits/1a69d3a8e583f63bc05efbc2055f9b510b4490d7)